### PR TITLE
Enable the orchestrator explorer by default

### PR DESCRIPTION
**Pull request by FlorianVeaux**
#### What this PR does / why we need it:
This PR enables the orchestrator explorer by default.
#### Special notes:

The important point of that PR is that it shouldn't introduce major incompatibilities for users, upgrading the helm chart shouldn't make the agent fail. As many variables comes into play, let's list the various scenarios.

Pre-requisites:
- Up to agent 7.24.x, the pod check on the process-agent requires that the `DD_ORCHESTRATOR_CLUSTER_ID` env variable is set. This is taken care of by our helm chart here: https://github.com/DataDog/helm-charts/blob/master/charts/datadog/templates/_container-process-agent.yaml#L38-L44
- With the next agent release (7.25.x), the pod check is now able to query the cluster agent to get the cluster_id. As long as the cluster agent is recent enough (future release 1.11), agent 7.25.x is able to live without the `DD_ORCHESTRATOR_CLUSTER_ID` env var.
- In the short-term, the helm chart will stop populating this environment variable: https://github.com/DataDog/helm-charts/pull/124 so the current PR should be able to work with this future change.

Decision tree:
Enable orchestration

In this diagram:

- `Config #1`: is when the user is explicitly defining both `clusterAgent.enabled` and `orchestratorExplorer.enabled`. In that case nothing happens with this PR and the values used are the ones provided by the user. Note that this can lead to enabling the orchestrator without the clusterAgent but that was already the case before.
- `Config #2`: is when the user explicitly sets `clusterAgent.enabled` to false and `orchestratorExplorer.enabled` is enabled (either via explicit configuration or via the new default). This is a new scenario where the orchestrator is enabled but can't work because `clusterAgent` is disabled. In that specific scenario, the helm chart disables the orchestrator explorer and logs a warning. Otherwise, the process-agent container would fail to start
- `Config #3`: is any other config basically.
- The red box `pod check is failing` means that on the process-agent container, the pod check will fail to get the cluster id and won't be able to run. In the end the orchestrator explorer page won't work at all for pods but the process-agent will continue to run.

#### Checklist
- [x] Documentation has been updated with helm-docs (run: `.github/helm-docs.sh`)
- [x] Chart Version bumped
- [x] `CHANGELOG.md` has beed updated
- [x] Variables are documented in the `README.md`

### CODE REVIEW

**Comment by vboulineau** [comment]
I'd expect this to go on top of all default review we have in:
https://github.com/DataDog/helm-charts/pull/100

**Review by hkaj — APPROVED** [review]
this is great! Let's merge as soon as we get approval from container-integrations :)

**Review by hkaj — APPROVED** [review]

**Review by vboulineau — APPROVED** [review]

**Review by clamoriniere — APPROVED** [review]

**Pull request merged**
